File: WebApplication/WebApplication/Controllers/RootController.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using WebApplication.DTO;
using WebApplication.Services;

namespace WebApplication.Controllers
{
    [ApiController]
    [Route("/")]
    public class RootController : ControllerBase
    {
        private UsersService _usersService;

        public RootController(UsersService usersService)
        {
            _usersService = usersService;
        }

        [HttpGet("")]
        public string GetRoot()
        {
            return "Root";
        }

        [HttpGet("string")]
        public string GetString()
        {
            return "Hello world!";
        }

        [HttpGet("json")]
        public UserDTO GetJson()
        {
            UserDTO userDto = _usersService.GetUser();
            return _usersService.GetUser();
        }

        [HttpGet("json-array")]
        public List<UserDTO> GetJsonArray()
        {
            return _usersService.GetUsers();
        }

        [HttpGet("json-array-task")]
        public Task<List<UserDTO>> GetJsonArrayTask()
        {
            return _usersService.GetUsersTask();
        }
    }
}

File: demo/src/main/java/com/example/demo/RootController.java
package com.example.demo;

import com.example.demo.dto.UserDTO;
import com.example.demo.services.UsersService;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
public class RootController {
    private UsersService usersService;

    public RootController(UsersService usersService) {
        this.usersService = usersService;
    }

    @RequestMapping(value = "/", method = RequestMethod.GET)
    public String index() {
        return "Root";
    }

    @RequestMapping(value = "/string", method = RequestMethod.GET)
    public String string() {
        return "Hello world!";
    }

    @RequestMapping(value = "/json", method = RequestMethod.GET)
    public UserDTO getJson() {
        return this.usersService.getUser();
    }

    @RequestMapping(value = "/json-array", method = RequestMethod.GET)
    public List<UserDTO> getJsonArray() {
        return this.usersService.getUsers();
    }
}
